activesupport/lib/active_support/callbacks.rb

```diff
@@ -663,7 +663,9 @@ def set_callback(name, *filter_list, &block)
           if options[:if].is_a?(String) || options[:unless].is_a?(String)
             ActiveSupport::Deprecation.warn(<<-MSG.squish)
               Passing string to :if and :unless conditional options is deprecated
-              and will be removed in Rails 5.2 without replacement.
+              and will be removed in Rails 5.2 without replacement. Instead of a
+              string for eval, pass a symbol for an instance method or an array
+              of symbols.
```

Comment:
You can also pass a lambda or a proc, or even a block. For that reason I prefer to not tell people what to do because no matter we try it will be always incomplete. If the problem is explain what the string is doing lets change it to: `Passing string to be evaluated in :if and :unless conditional options is deprecated`